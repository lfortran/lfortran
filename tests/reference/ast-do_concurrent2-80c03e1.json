{
    "basename": "ast-do_concurrent2-80c03e1",
    "cmd": "lfortran --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/do_concurrent2.f90",
    "infile_hash": "afe8788676d4b7be4b765236db5d9c60aa267325a08b714b35ba7eb6",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-do_concurrent2-80c03e1.stdout",
    "stdout_hash": "845f1304d620c74eed27b4127b486a19cdd0429f5f75df06c92ba036",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}