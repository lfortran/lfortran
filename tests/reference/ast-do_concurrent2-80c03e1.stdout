(TranslationUnit
    [(Subroutine
        do_concurrent2
        [(a)
        (b)]
        []
        ()
        ()
        []
        []
        []
        [(Declaration
            (AttrType
                TypeReal
                []
                ()
                ()
                None
            )
            [(AttrIntent
                InOut
            )]
            [(a
            [(()
            ()
            DimensionExpr)]
            []
            ()
            None
            ())
            (b
            [(()
            ()
            DimensionExpr)]
            []
            ()
            None
            ())]
            ()
        )
        (Declaration
            (AttrType
                TypeInteger
                []
                ()
                ()
                None
            )
            []
            [(i
            []
            []
            ()
            None
            ())]
            ()
        )
        (Declaration
            (AttrType
                TypeReal
                []
                ()
                ()
                None
            )
            []
            [(x
            []
            []
            ()
            None
            ())]
            ()
        )]
        [(=
            0
            x
            (Real "1.0")
            ()
        )
        (DoConcurrentLoop
            0
            ()
            [(ConcurrentControl
                i
                1
                10
                ()
            )]
            ()
            [(ConcurrentShared
                [i]
            )
            (ConcurrentLocal
                [x]
            )
            (ConcurrentDefault)]
            [(If
                0
                ()
                (> (FuncCallOrArray
                    a
                    []
                    [(()
                    i
                    ()
                    0)]
                    []
                    []
                    []
                ) 0)
                [(=
                    0
                    x
                    (FuncCallOrArray
                        sqrt
                        []
                        [(()
                        (FuncCallOrArray
                            a
                            []
                            [(()
                            i
                            ()
                            0)]
                            []
                            []
                            []
                        )
                        ()
                        0)]
                        []
                        []
                        []
                    )
                    ()
                )
                (=
                    0
                    (FuncCallOrArray
                        a
                        []
                        [(()
                        i
                        ()
                        0)]
                        []
                        []
                        []
                    )
                    (- (FuncCallOrArray
                        a
                        []
                        [(()
                        i
                        ()
                        0)]
                        []
                        []
                        []
                    ) (** x 2))
                    ()
                )]
                []
                ()
                ()
                ()
            )
            (=
                0
                (FuncCallOrArray
                    b
                    []
                    [(()
                    i
                    ()
                    0)]
                    []
                    []
                    []
                )
                (- (FuncCallOrArray
                    b
                    []
                    [(()
                    i
                    ()
                    0)]
                    []
                    []
                    []
                ) (FuncCallOrArray
                    a
                    []
                    [(()
                    i
                    ()
                    0)]
                    []
                    []
                    []
                ))
                ()
            )]
            ()
            ()
        )
        (Print
            0
            ()
            [x]
            ()
        )]
        []
        []
    )]
)
